src/daft-logical-plan/src/stats.rs

```diff
@@ -0,0 +1,121 @@
+use std::fmt::Display;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq, Hash)]
+pub enum StatsState {
+    Materialized(PlanStats),
+    NotMaterialized,
+}
+
+impl StatsState {
+    pub fn unwrap_or_default(&self) -> PlanStats {
```

Comment:
this should take ownership and skip the clone.

Reply:
Removed this since we only really need `materialized_stats()`